src/error.rs

```diff
@@ -22,8 +25,10 @@ pub enum Error {
     /// An `io::Error` that occurred while trying to read or write to a network stream.
     Io(IoError),
     /// An error from the `native_tls` library during the TLS handshake.
+    #[cfg(feature = "default")]
```

Comment:
So, this is actually pretty problematic, because it makes features non-additive. I think for this to be okay, we'll need https://github.com/rust-lang/rust/issues/44109, which hasn't landed yet. One possible solution would be to add a variant to the end of this `enum` like this:
```rust
#[doc(hidden)]
__Nonexhaustive,
```
That prevents people from matching on `Error`, and thus makes the features additive again.